# occupancy grid information
world_collision_model:  # Currently it's taken fron ur5.yaml. Is it correct?
  frame_id: base_link
  size_x: 1.9
  size_y: 2.4
  size_z: 3.0
  origin_x: -0.4
  origin_y: -1.2
  origin_z: 0.0
  res_m: 0.02
  max_distance_m: 0.2

robot_collision_model:
  world_joint:
    name: world
    type: planar # fixed, floating
  voxels_models:
    - { link_name: base_link_inertia, res: 0.01 } # base_link_inertia?
    - { link_name: shoulder_link, res: 0.01 }
    - { link_name: upper_arm_link, res: 0.01 }
    - { link_name: forearm_link, res: 0.01 }
    - { link_name: wrist_1_link, res: 0.01 }
    - { link_name: wrist_2_link, res: 0.01 }
    - { link_name: wrist_3_link, res: 0.01 }
  spheres_models:
    - link_name: shoulder_link
      auto: false
      spheres:
        - { name: sh0, x: 0.0, y: 0.000, z: 0.00, radius: 0.10, priority: 5 }
    - link_name: upper_arm_link
      auto: false
      #      If true then uncomment the next line:
#      radius: 0.01
      spheres:
        - { name: ua0, x: -0.13, y: 0.0, z: 0.18, radius: 0.07, priority: 5 }
        - { name: ua1, x: -0.26, y:  0.0, z:  0.18, radius: 0.07, priority: 4 }
        - { name: ua2, x: -0.39, y:  0.0, z: 0.18, radius: 0.07, priority: 4 }
        - { name: ua3, x: -0.52, y:  0.0, z:  0.18, radius: 0.07, priority: 4 }
    - link_name: forearm_link
      auto: false
      #      If true then uncomment the next line:
#      radius: 0.0045
      spheres:
        - { name: fa0, x: 0.0, y:  0.0, z:  0.04, radius: 0.07, priority: 2 }
        - { name: fa1, x: -0.15, y:  0.0, z:  0.04, radius: 0.07, priority: 2 }
        - { name: fa2, x: -0.30, y:  0.0, z:  0.04, radius: 0.07, priority: 2 }
        - { name: fa3, x: -0.45, y:  0.0, z:  0.04, radius: 0.07, priority: 2 }
        - { name: fa4, x: -0.58, y: 0.0, z:  0.04, radius: 0.05, priority: 2 }
    - link_name: wrist_1_link
      auto: false
      #      If true then uncomment the next line:
#      radius: 0.0050
      spheres:
        - { name: w10, x: 0.0, y:  0.0, z: 0.0, radius: 0.06, priority: 2 }
    - link_name: wrist_2_link
      auto: false
      #      If true then uncomment the next line:
#      radius: 0.0045
      spheres:
        - { name: w10, x: 0.0, y: 0.0, z: 0.0, radius: 0.06, priority: 1 }
#    - link_name: wrist_3_link
#      auto: false
#      #      If true then uncomment the next line:
#      radius: 0.0045

#  spheres_models:
#    - link_name: base_link_inertia
#      auto: true
#      radius: 0.01
#    - link_name: shoulder_link
#      auto: true
#      #      If true then uncomment the next line:
#      radius: 0.01
##      spheres:
##        - { name: sh0, x: 0.0, y:  0.000, z:  0.000, radius: 0.080, priority: 5 }
##        - { name: sh1, x: 0.0, y:  0.070, z:  0.000, radius: 0.080, priority: 5 }
#    - link_name: upper_arm_link
#      auto: true
#      #      If true then uncomment the next line:
#      radius: 0.01
##      spheres:
##        - { name: ua0, x: 0.0, y:  0.000, z:  0.000, radius: 0.080, priority: 4 }
##        - { name: ua1, x: 0.0, y:  0.000, z:  0.105, radius: 0.080, priority: 4 }
##        - { name: ua2, x: 0.0, y:  0.000, z:  0.210, radius: 0.080, priority: 4 }
##        - { name: ua3, x: 0.0, y:  0.000, z:  0.315, radius: 0.080, priority: 2 }
##        - { name: ua4, x: 0.0, y:  0.000, z:  0.420, radius: 0.080, priority: 2 }
#    - link_name: forearm_link
#      auto: true
#      #      If true then uncomment the next line:
#      radius: 0.0045
##      spheres:
##        - { name: fa0, x: 0.0, y:  0.000, z:  0.000, radius: 0.045, priority: 4 }
##        - { name: fa1, x: 0.0, y:  0.000, z:  0.100, radius: 0.045, priority: 4 }
##        - { name: fa2, x: 0.0, y:  0.000, z:  0.200, radius: 0.045, priority: 4 }
##        - { name: fa3, x: 0.0, y:  0.000, z:  0.300, radius: 0.045, priority: 2 }
##        - { name: fa4, x: 0.0, y: -0.019, z:  0.390, radius: 0.045, priority: 2 }
##        - { name: fa5, x: 0.0, y:  0.019, z:  0.390, radius: 0.045, priority: 2 }
#    - link_name: wrist_1_link
#      auto: true
#      #      If true then uncomment the next line:
#      radius: 0.0050
##      spheres:
##        - { name: w10, x: 0.0, y:  0.090, z: -0.018, radius: 0.050, priority: 2 }
##        - { name: w11, x: 0.0, y:  0.090, z:  0.018, radius: 0.050, priority: 2 }
#    - link_name: wrist_2_link
#      auto: true
#      #      If true then uncomment the next line:
#      radius: 0.0045
#
#    - link_name: wrist_3_link
#      auto: true
#      #      If true then uncomment the next line:
#      radius: 0.0045


#      spheres:
#        - { name: w20, x: 0.0, y: -0.018, z:  0.090, radius: 0.045, priority: 1 }
#        - { name: w21, x: 0.0, y:  0.018, z:  0.090, radius: 0.045, priority: 1 }
        #      If true then uncomment the next line:

#    - link_name: ee_link
#  voxels_models:
#    - link_name: world
  collision_groups:
    - name: manipulator
      links:
        - name: base_link_inertia
        - name: shoulder_link
        - name: upper_arm_link
        - name: forearm_link
        - name: wrist_1_link
        - name: wrist_2_link
        - name: wrist_3_link
        - name: flange
#allowed_collisions:
#  - { sphere1: sh0, sphere2: ua0 }
#  - { sphere1: sh1, sphere2: ua0 }
#  - { sphere1: sh1, sphere2: ua1 }
#  - { sphere1: ua4, sphere2: fa0 }
#  - { sphere1: fa5, sphere2: w10 }
#  - { sphere1: fa5, sphere2: w11 }
#  - { sphere1: w11, sphere2: w20 }
#  - { sphere1: w11, sphere2: w21 }
